packages/salesforcedx-vscode-core/test/jest/settings/sfdxCoreSettings.test.ts

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { ENABLE_SOURCE_TRACKING_FOR_DEPLOY_RETRIEVE } from '../../../src/constants';
+import { SfdxCoreSettings } from '../../../src/settings/sfdxCoreSettings';
+
+describe('sfdxCoreSettings', () => {
+  const sfdxCoreSettingsInstance = SfdxCoreSettings.getInstance();
+  let getConfigValueSpy: jest.SpyInstance;
+  getConfigValueSpy = jest.spyOn((SfdxCoreSettings as any).prototype, 'getConfigValue');
+  describe('getEnableSourceTrackingForDeployAndRetrieve', () => {
+    it('should set the default value for enable source tracking to be true', () => {
```

Comment:
nit: this test title is a little confusing.  I'd go with, `Should return the value for enable source tracking  with default of true.`

Reply:
Also here you can setup the spy to return a value so you can verify that the function returned what you expected
```
getConfigValueSpy.mockReturnsValue(true);
const result = sfdxCoreSettingsInstance.getEnableSourceTrackingForDeployAndRetrieve();
expect(result).toEqual(true);
... the other expects
```

Reply:
This function is setting the default value to be true.

Reply:
isn the main purpose to get the current setting (with a default of true)? hence the function name `getEnableSourceTrackingForDeployAndRetrieve`

Reply:
Yes! The current setting is true by default. And this is where we are setting its default value to be true by passing `true` as second parameter in the function `getConfigValue`.